ast/term.go

```diff
@@ -1681,7 +1683,9 @@ func (s *set) insert(x *Term) {
 	s.elems[insertHash] = x
 	// O(1) insertion, but we'll have to re-sort the keys later.
 	s.keys = append(s.keys, x)
-	s.numInserts++ // Track insertions since the last re-sorting.
+	// Reset the sync.Once instance.
+	// See https://github.com/golang/go/issues/25955 for why we do it this way.
+	s.sortGuard = new(sync.Once)
```

Comment:
For my understanding: we do this more often that we need to, but that's not a problem, the penalty is low. We need to do this more often because there's no way to tell if a given sync.Once (`s.sortGuard`) has been discharged or not.

Reply:
Right. During benchmarks, this doesn't seem to cause spurious allocs, so I think the compiler is catching our intent (and may just have an atomics-guarded boolean or something under-the-hood).